fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -5789,7 +5790,8 @@ public void processPostDisbursementTransactions() {
 
     private LoanScheduleDTO getRecalculatedSchedule(final ScheduleGeneratorDTO generatorDTO) {
 
-        if (!this.repaymentScheduleDetail().isInterestRecalculationEnabled() || isNpa || isChargedOff()) {
+        if (!this.repaymentScheduleDetail().isEnableDownPayment()
```

Comment:
Please check this also.

Reply:
after separating the down payment from interest recalculation, I still need this check at this point